pkg/experiment/compactor/compaction_worker.go

```diff
@@ -63,6 +68,12 @@ func (cfg *Config) Validate() error {
 }
 
 func New(config Config, logger log.Logger, metastoreClient *metastoreclient.Client, storage objstore.Bucket, reg prometheus.Registerer) (*Worker, error) {
+	undo, err := maxprocs.Set(maxprocs.Logger(log2.Printf))
+	if err != nil {
+		_ = level.Error(logger).Log("msg", "failed to set maxprocs", "err", err)
+		undo()
+	}
+	workers := runtime.GOMAXPROCS(-1) * config.JobConcurrency
```

Comment:
It's safe to simply import `go.uber.org/automaxprocs` for side effects.

As far as I can judge by [the source](https://github.com/uber-go/automaxprocs/blob/1ea14c35ce47a73089b824e504d1c92eeb61a5a6/maxprocs/maxprocs.go#L115-L117), `undo` is only valid if we did change `GOMAXPROCS`:
```go
	if err != nil {
		return undoNoop, err
	}
```

And if an error is encountered, it will be logged (from [`automaxprocs`](https://github.com/uber-go/automaxprocs/blob/master/automaxprocs.go)):
```go
func init() {
	maxprocs.Set(maxprocs.Logger(log.Printf))
}
```

Reply:
> It's safe to simply import go.uber.org/automaxprocs for side effects.

Yeah, I wanted to touch only compactor. Let's touch everything.

Reply:
Ah, I see now – `go.uber.org/automaxprocs` has led me astray. I personally think that `automaxprocs` won't hurt us if used globally